Source/Redemption/Private/PowerUpComponent.cpp

```diff
@@ -83,4 +87,24 @@ void UPowerUpComponent::ChangeActivePowerUp(EActivePowerUp NewPower)
 {
 	CurrentPowerUp = NewPower;
 	ChangeMappingContext();
+}
+
+//will spawn the bone
+void UPowerUpComponent::ExecuteBonePowerUp()
+{
+	if (BoneBP)
+	{
+		if (ACharacter* OwnerChar = Cast<ACharacter>(GetOwner()))
+		{
+			FVector FowardVec = OwnerChar->GetActorForwardVector();
+			FVector SpawnPos = OwnerChar->GetActorLocation() + FowardVec * 100;
+
+			FRotator CharacterRotation = FowardVec.Rotation();
+
+			FVector BoneDirection = FowardVec + FVector(0.0, 0.0, 0.3);
+
+			ABone* Bone = GetWorld()->SpawnActor<ABone>(BoneBP, SpawnPos, FRotator::ZeroRotator);
+			Bone->GetMesh()->AddImpulse(BoneDirection * BONE_SPEED);
```

Comment:
In the comment, you said that this code will spawn a bone, but it seems like throwing a bone. If this is about throwing in a bone in front of the player, we can't set the destination of the bone right?

Reply:
yes you are right it spawns and throws the bone. at the moment the bone will always move in front of the player. I will change that when we have the first perspective

Reply:
Okay, thank you. I can't wait to see that